plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts

```diff
@@ -167,12 +166,28 @@ export class DatabaseEntitiesCatalog implements EntitiesCatalog {
             const context = { kind, namespace, locationId };
             for (let attempt = 1; attempt <= BATCH_ATTEMPTS; ++attempt) {
               try {
-                const { toAdd, toUpdate } = await this.analyzeBatch(
+                const { toAdd, toUpdate, toIgnore } = await this.analyzeBatch(
                   batch,
                   context,
                 );
-                if (toAdd.length) await this.batchAdd(toAdd, context);
-                if (toUpdate.length) await this.batchUpdate(toUpdate, context);
+                if (toAdd.length) {
+                  modifiedEntityIds.push(
```

Comment:
Nop, would have to refactor this into a list of operations instead of lists for each operation. Maybe worthwhile, but followup PR?

Reply:
Followup.

Can also return a full_name->id dict that the caller can make lookups from if they want to or similar so we don't have to worry about ordering. I do worry a lot about performance in these parts, because I've already seen them matter way more and way sooner than I had anticipated. We don't want to risk hurting performance by breaking apart batches or similar just to fulfil the ordering thing :)